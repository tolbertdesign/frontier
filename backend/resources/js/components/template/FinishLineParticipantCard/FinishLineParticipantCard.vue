<template>
  <div class="max-w-lg mx-auto p-4 my-2 border border-grey-light rounded-lg shadow">
    <div class="media p-4">
      <div class="media-left">
        <AvatarImage
          :alt="participant.first_name"
          :src="avatarSrc"
        />
      </div>
      <div class="media-content self-center text-xl">
        <h2 class="text-base sm:text-xl font-dark text-black">
          {{ participant.first_name }}
        </h2>
        <h3 class="sm:text-sm text-xs mb-2 capitalize">
          {{ unit.name_plural }}
        </h3>
      </div>
      <div class="media-right pt-2 self-start has-text-right text-base sm:text-2xl">
        <UnitsForm
          :participant-user-id="participant.id"
          :units="participant.laps"
          :unit-name-plural="unit.name_plural"
          :unit-max="unitMax"
          :can-edit="canEdit"
        />
      </div>
    </div>
  </div>
</template>

<script>
import UnitsForm from '@/components/template/UnitsForm'
import AvatarImage from '@/components/element/AvatarImage'

export default {
  name: 'FinishLineParticipantCard',
  status: 'prototype',
  release: '1.0.0',
  components: {
    UnitsForm,
    AvatarImage,
  },
  props: {
    participant: {
      type: Object,
      default: () => {},
    },
    unit: {
      type: Object,
      default: () => {},
    },
    unitMax: {
      type: Number,
      default: 35,
    },
    canEdit: {
      type: Boolean,
      default: true,
    },
  },
  computed: {
    lang () {
      return this.$store.state.lang
    },
    avatarSrc () {
      return this.participant.profile.image_name == null ? '/v3-assets/dashboard/images/dashboard-avatar.svg' : this.avatarImgSrc
    },
    avatarImgSrc () {
      return this.$store.getters.avatarPath + this.participant.profile.image_name
    },
  },
}
</script>
