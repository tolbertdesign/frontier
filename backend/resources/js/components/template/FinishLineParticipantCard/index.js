import FinishLineParticipantCard from './FinishLineParticipantCard'

export default FinishLineParticipantCard
