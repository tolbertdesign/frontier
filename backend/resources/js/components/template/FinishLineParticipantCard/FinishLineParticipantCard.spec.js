import { shallowMount, createLocalVue } from '@vue/test-utils'
import FinishLineParticipantCard from './FinishLineParticipantCard.vue'

const localVue = createLocalVue()

describe('FinishLineParticipantCard', () => {
  const participant = {
    id: 5,
    first_name: 'Jane',
    last_name: 'Doe',
    profile: {
      image_name: 'test-image',
    },
  }

  const unit = {
    name_plural: 'Name Plural',
  }

  const $store = {
    getters: {
      avatarPath: 'avatar-path',
    },
  }

  it('renders finish line participant card correctly', () => {
    const wrapper = shallowMount(FinishLineParticipantCard, {
      mocks: { $store },
      propsData: {
        participant,
        unit,
        unitMax: 30,
      },
      localVue,
    })
    expect(wrapper).toMatchSnapshot()
  })
})
