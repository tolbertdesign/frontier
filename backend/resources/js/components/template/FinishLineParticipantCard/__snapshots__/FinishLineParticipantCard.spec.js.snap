// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinishLineParticipantCard renders finish line participant card correctly 1`] = `
<div class="max-w-lg mx-auto p-4 my-2 border border-grey-light rounded-lg shadow">
  <div class="media p-4">
    <div class="media-left">
      <avatarimage-stub alt="Jane" src="avatar-pathtest-image"></avatarimage-stub>
    </div>
    <div class="media-content self-center text-xl">
      <h2 class="text-base sm:text-xl font-dark text-black">
        Jane
      </h2>
      <h3 class="sm:text-sm text-xs mb-2 capitalize">
        Name Plural
      </h3>
    </div>
    <div class="media-right pt-2 self-start has-text-right text-base sm:text-2xl">
      <unitsform-stub participantuserid="5" units="0" unitnameplural="Name Plural" unitmax="30" canedit="true"></unitsform-stub>
    </div>
  </div>
</div>
`;
